File: parental_src.py
import time

def isNaN(string):
    if(type(string) != str):
        return False if type(string) == int else True
    else:
        try:
            int(string)

            return False
        except:
            return True

class CRC32:
    def __init__(self):
        self.gentable()

    def crc32(self, input = "", crc = 0xFFFFFFFF):
        count = len(input)

        i = 0

        while(count != 0):
            count -= 1

            temp1 = (crc >> 8) & 0xFFFFFF
            temp2 = self.table[(crc ^ ord(input[i])) & 0xFF]

            crc = temp1 ^ temp2

            i += 1
        
        return crc
    
    def gentable(self):
        self.table = []

        for i in range(256):
            crc = i

            for j in range(8):
                if(crc & 1):
                    crc = (crc >> 1) ^ 0xEDB88320
                else:
                    crc >>= 1
            
            self.table.append(crc)

def process(num):
    if(type(num) != str or len(num) != 8 or isNaN(num)):
        return None
    else:
        return ((CRC32().crc32((time.strftime("%m%d", time.gmtime(time.time())) + num[4:8])) ^ 0xAAAA) + 0x14C1) % 100000

print(process("27281938"))

File: parental_src.js
class CRC32 {
    constructor(){
        this.gentable()
    }

    crc32(input = "", crc = 0xFFFFFFFFn){
        var count = input.length

        var i = 0

        while(count !== 0){
            count--

            var temp1 = (crc >> 8n) & 0xFFFFFFn
            var temp2 = this.table[(crc ^ BigInt(input.charCodeAt(i))) & 0xFFn]

            crc = temp1 ^ temp2

            i++
        }

        return crc
    }

    gentable(){
        this.table = []

        for(var i = 0; i < 256; i++){
            var crc = BigInt(i)

            for(var j = 0; j < 8; j++){
                if(crc & 1n){
                    crc = (crc >> 1n) ^ 0xEDB88320n
                } else {
                    crc >>= 1n
                }
            }

            this.table.push(crc)
        }
    }
}

const isInvalidInteger = (anyInt) => (isNaN(anyInt) || anyInt.toString().includes("e") || !isFinite(anyInt) || !Number.isInteger(anyInt) || !Number.isSafeInteger(anyInt))

function fixedInt(int = 0, length = 0){
    if([int, length].some(k => isInvalidInteger(k))) return null

    int = int.toString()

    var l = length - int.length

    return l <= 0 ? int : "0".repeat(l) + int
}

function process(num){
    if(typeof num !== "string" || num.length !== 8 || isNaN(num)) return null

    const curDate = new Date()

    return BigInt((new CRC32().crc32((`${fixedInt(curDate.getMonth() + 1, 2)}${fixedInt(curDate.getDate(), 2)}` + num.slice(4, 8))) ^ 0xAAAAn) + 0x14C1n) % 100000n
}

console.log(parseInt(process("27281938")))